core/app/finders/spree/products/find.rb

```diff
@@ -170,18 +170,30 @@ def by_option_value_ids(products)
       end
 
       def by_properties(products)
-        return products unless properties? && properties.values.reject(&:empty?).present?
+        return products unless properties?
 
-        product_ids = properties.to_unsafe_hash.map do |property_filter_param, product_properties_values|
+        product_ids = []
+        index = 0
+
+        properties.to_unsafe_hash.each do |property_filter_param, product_properties_values|
           next if property_filter_param.blank? || product_properties_values.empty?
 
           values = product_properties_values.split(',').map(&:parameterize)
 
-          products.
-            joins(product_properties: :property).
-            where(spree_properties: { filter_param: property_filter_param.parameterize }).
-            where(spree_product_properties: { filter_param: values }).ids
-        end.flatten.compact.uniq
+          next if values.empty?
+
+          ids = products.
+                joins(product_properties: :property).
+                where(spree_properties: { filter_param: property_filter_param.parameterize }).
+                where(spree_product_properties: { filter_param: values }).distinct.ids
+
+          if index == 0
```

Comment:
What do you think about moving this out of the loop and finding the intersection between product ids after finding them (with `map`):
```
product_ids.reduce(&:intersection)
```

Where `product_ids` would be eg. `[[1, 2], [2, 3, 4], [1, 2, 3]]`.

I know that, in the end, it's the same thing. But I find reducing the array afterwards more clear when reading the code.